### PR TITLE
ISPN-16537 Cache health and config do not respond 404 for non running caches

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-16537

### CODE REVIEW

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java [hunk 1]
```diff
@@ -967,17 +968,9 @@ private CompletionStage<RestResponse> getCacheConfig(RestRequest request) {
...
-      AuthorizationManager authorizationManager = SecurityActions.getCacheAuthorizationManager(cache.getAdvancedCache());
-      if (authorizationManager == null || authorizationManager.isPermissive()) {
-         // Cache is not secured, use the global authz
-         invocationHelper.getRestCacheManager().getAuthorizer().checkPermission(AuthorizationPermission.ADMIN);
-      }
-      Configuration cacheConfiguration = cache.getCacheConfiguration();
-
+      EmbeddedCacheManager cacheManager = invocationHelper.getRestCacheManager().getInstance();
+      EmbeddedCacheManager subjectCacheManager = cacheManager.withSubject(request.getSubject());
+      Configuration cacheConfiguration = SecurityActions.getCacheConfiguration(subjectCacheManager, cacheName);
```
Comment (fax4ever) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:973]: Do we want provide cache configurations also to non admin?

**Review by fax4ever — COMMENTED** [review]
Just a question. I'm out of my field here... trying to review it :P

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java (hunk 1 above)
Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:973]: @fax4ever no, and we aren't providing it. the endpoint is secured for admins

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java [hunk 2]
```diff
@@ -178,15 +177,17 @@ public Invocations getInvocations() {
...
             .invocation().methods(GET).path("/v2/caches/{cacheName}").withAction("listen").handleWith(this::cacheListen)
 
-            // Config and statistics
-            .invocation().methods(GET, HEAD).path("/v2/caches/{cacheName}").withAction("config").handleWith(this::getCacheConfig)
-            .invocation().methods(GET).path("/v2/caches/{cacheName}").withAction("stats").handleWith(this::getCacheStats)
-            .invocation().methods(POST).path("/v2/caches/{cacheName}").withAction("stats-reset").permission(AuthorizationPermission.ADMIN).handleWith(this::resetCacheStats)
-            .invocation().methods(GET).path("/v2/caches/{cacheName}").withAction("distribution").handleWith(this::getCacheDistribution)
+            // Config
+            .invocation().methods(GET, HEAD).path("/v2/caches/{cacheName}").withAction("config").permission(AuthorizationPermission.ADMIN).handleWith(this::getCacheConfig)
             .invocation().methods(GET).path("/v2/caches/{cacheName}").withAction("get-mutable-attributes").permission(AuthorizationPermission.ADMIN).handleWith(this::getCacheConfigMutableAttributes)
```
Comment (karesti) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:182]: @fax4ever here

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java (hunk 1 above)
Comment (fax4ever) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:973]: you're right... I didn't get it

**Comment by karesti** [comment]
@tristantarrant I relaunched the build because it was broken with more than 1000 failures

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**